src/corehost/cli/ijwhost/ijwhost.cpp

```diff
@@ -30,14 +30,53 @@ pal::hresult_t get_load_in_memory_assembly_delegate(pal::dll_t handle, load_in_m
             pal::string_t mod_path;
             if (!pal::get_module_path(handle, &mod_path))
             {
-                trace::error(_X("Failed to resolve full path of the current mixed-mode module [%s]"), host_path.c_str());
+                trace::error(_X("Failed to resolve full path of the current mixed-mode module [%s]"), mod_path.c_str());
                 return StatusCode::LibHostCurExeFindFailure;
             }
 
             pal::string_t config_path_local { strip_file_ext(mod_path) };
             config_path_local.append(_X(".runtimeconfig.json"));
+
             *config_path_out = std::move(config_path_local);
 
+            // if file does not exist, try to get the one used for an existing context (if there is any)
+            if (!pal::realpath(&config_path_local))
+            {
+                trace::info(_X("Failed to resolve .runtimeconfig.json path of the current mixed-mode module [%s]"), mod_path.c_str());
+                trace::info(_X("Trying to get .runtimeconfig.json from existing context"));
+
+                pal::dll_t out_fxr;
+                pal::string_t out_fxr_path;
+
+                if (pal::get_loaded_library(LIBFXR_NAME, "hostfxr_get_runtime_property_value", &out_fxr, &out_fxr_path))
+                {
+                    hostfxr_get_runtime_property_value_fn get_runtime_property_value = reinterpret_cast<hostfxr_get_runtime_property_value_fn>(pal::get_symbol(out_fxr, "hostfxr_get_runtime_property_value"));
```

Comment:
If the type is properly declared on the right side, feel free to use `auto` on the left. These function `typedef`s and names tend to get out of control.